storage/pin/pin.go

```diff
@@ -0,0 +1,408 @@
+	}
+
+	log.Debug("File unpinned", "Address", rootHash)
+	return nil
+}
+
+// ListPinFiles functions logs information of all the files that are pinned
+// in the current local node. It displays the root hash of the pinned file
+// or collection. It also display three vital information's
+//     1) Whether the file is a RAW file or not
+//     2) Size of the pinned file or collection
+//     3) the number of times that particular file or collection is pinned.
+func (p *API) ListPinFiles() map[string]FileInfo {
+	p.pinLock.RLock()
+	defer p.pinLock.RUnlock()
+	for hash, fileInfo := range p.pinnedFiles {
+		log.Info("Pinned file", "Address", hash, "IsRAW", fileInfo.IsRaw,
+			"FileSize", fileInfo.FileSize, "PinCounter", fileInfo.FileSize)
+	}
+	return p.pinnedFiles
+}
+
+func (p *API) walkChunksFromRootHash(rootHash string, isRaw bool, credentials string, executeFunc func(storage.Reference) error) {
+	fileWorkers := make(chan storage.Reference, WorkerChanSize)
+	chunkWorkers := make(chan storage.Reference, WorkerChanSize)
+
+	addr, err := hex.DecodeString(rootHash)
+	if err != nil {
+		log.Error("Error decoding root hash", "err", err)
+		return
+	}
+
+	hashFunc := storage.MakeHashFunc(storage.DefaultHash)
+	hashSize := len(addr)
+	isEncrypted := len(addr) > hashFunc().Size()
+	getter := storage.NewHasherStore(p.db, hashFunc, isEncrypted, chunk.NewTag(0, "show-chunks-tag", 0))
+
+	go func() {
+		if !isRaw {
+
+			// If it is not a raw file... load the manifest and process the files inside one by one
+			walker, err := p.api.NewManifestWalker(context.Background(), storage.Address(addr),
+				p.api.Decryptor(context.Background(), credentials), nil)
+
+			if err != nil {
+				log.Error("Could not decode manifest.", "err", err)
+				return
+			}
+
+			err = walker.Walk(func(entry *api.ManifestEntry) error {
+
+				fileAddr, err := hex.DecodeString(entry.Hash)
+				if err != nil {
+					log.Error("Error decoding hash present in manifest", "err", err)
+					return err
+				}
+
+				// send the file to file workers
+				fileWorkers <- storage.Reference(fileAddr)
+
+				return nil
+			})
+
+			if err != nil {
+				log.Error("Error walking manifest", "err", err)
+				return
+			}
+
+			// Finally, remove the manifest file too
+			fileWorkers <- storage.Reference(addr)
+
+			// Signal end of file stream
+			close(fileWorkers)
+
+		} else {
+			// Its a raw file.. no manifest.. so process only this hash
+			fileWorkers <- storage.Reference(addr)
+
+			// Signal end of file stream
+			close(fileWorkers)
+
+		}
+	}()
+
+	for fileRef := range fileWorkers {
+
+		// Send the file to chunk workers
+		chunkWorkers <- fileRef
+
+		actualFileSize := uint64(0)
+		rcvdFileSize := uint64(0)
+		doneChunkWorker := make(chan struct{})
+		var cwg sync.WaitGroup // Wait group to wait for chunk processing to complete
+
+	QuitChunkFor:
+		for {
+			select {
+			case <-doneChunkWorker:
+				break QuitChunkFor
+
+			case ref := <-chunkWorkers:
+				cwg.Add(1)
+
+				go func() {
+
```

Comment:
rm ws

Reply:
Is it not the same place as above comment?